File: app/src/main/java/com/loopeer/codereader/ui/adapter/ItemTouchHelperCallback.java
package com.loopeer.codereader.ui.adapter;

import android.graphics.Canvas;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;

import com.loopeer.itemtouchhelperextension.ItemTouchHelperExtension;

public class ItemTouchHelperCallback extends ItemTouchHelperExtension.Callback {

    @Override
    public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
        return makeMovementFlags(0, ItemTouchHelper.START);
    }

    @Override
    public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
        return false;
    }

    @Override
    public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {

    }

    @Override
    public void onChildDraw(Canvas c, RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, float dX, float dY, int actionState, boolean isCurrentlyActive) {
        if (viewHolder instanceof MainLatestAdapter.RepoViewHolder)
            ((MainLatestAdapter.RepoViewHolder) viewHolder).mProgressRelativeLayout.setTranslationX(dX);
    }
}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/ui/adapter/ItemTouchHelperCallback.kt
package com.loopeer.codereaderkt.ui.adapter

import android.graphics.Canvas
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.helper.ItemTouchHelper
import com.loopeer.itemtouchhelperextension.ItemTouchHelperExtension


class ItemTouchHelperCallback : ItemTouchHelperExtension.Callback() {

    override fun getMovementFlags(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder): Int {
        return ItemTouchHelperExtension.Callback.makeMovementFlags(0, ItemTouchHelper.START)
    }

    override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
        return false
    }

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {

    }

    override fun onChildDraw(c: Canvas, recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, dX: Float, dY: Float, actionState: Int, isCurrentlyActive: Boolean) {
//        if (viewHolder is MainLatestAdapter.RepoViewHolder)
//            (viewHolder as MainLatestAdapter.RepoViewHolder).mProgressRelativeLayout.setTranslationX(dX)
    }
}
